src/react-native/06-building-custom-components/building-custom-components.md

```diff
@@ -39,10 +44,136 @@ If you’ve implemented the solution correctly, the tests will pass when you run
 <details>
 <summary>Click to see the solution</summary>
 
+✏️ Update **App.tsx** to be:
+
+@diff ../../../exercises/react-native/06-custom-components/01-problem/App.tsx ../../../exercises/react-native/06-custom-components/01-solution/App.tsx only
+
+</details>
+
+## Objective 2: Create component type properties
+
+TODO
+
+### Concept TODO
+
+TODO
+
+### Setup 2
+
+✏️ Update **App.tsx** to be:
+
+@diff ../../../exercises/react-native/06-custom-components/01-solution/App.tsx ../../../exercises/react-native/06-custom-components/02-problem/App.tsx only
+
+exercises/react-native/06-custom-components/01-problem/App.tsx
+
+### Verify 2
+
+✏️ Update **App.test.tsx** to be:
+
+@diff ../../../exercises/react-native/06-custom-components/01-solution/App.test.tsx ../../../exercises/react-native/06-custom-components/02-problem/App.test.tsx only
+
+### Exercise 2
+
+- Update the `ListItemProps` type to enforce a `key` and `name` property of the appropriate primitive type.
+- Update the `ListItem` component to use the `ListItemProps` as its input type and return the `name` property of the a `state` index.
+- Update the `StateList` component to use the `ListItem` component handle each state index.
+
+### Solution 2
+
+If you’ve implemented the solution correctly, the tests will pass when you run `npm run test`!
+
+<details>
+<summary>Click to see the solution</summary>
+
+✏️ Update **App.tsx** to be:
+
+@diff ../../../exercises/react-native/06-custom-components/02-problem/App.tsx ../../../exercises/react-native/06-custom-components/02-solution/App.tsx
+
+
+</details>
+
+## Objective 3: The Fractal Modlet Pattern 
+
 TODO
 
+### Concept TODO
+
+TODO
+
+### Setup 3
+
+It’s best practice to create a new folder that will contain all of the related files for each component, including test files.
+
+✏️ Update **index.js** to be:
+
+@diff ../../../exercises/react-native/06-custom-components/02-solution/index.js ../../../exercises/react-native/06-custom-components/03-problem/index.js
+
+✏️ Create src/ (folder)
+
+✏️ Move **App.tsx** to **/src/App.tsx** and update it to be:
+
+@diff ../../../exercises/react-native/06-custom-components/02-solution/App.tsx ../../../exercises/react-native/06-custom-components/03-problem/src/App.tsx
```

Comment:
Be sure to use `only` on these diffs so they’re not as large.

Or, if there is no difference in the files and they’re just moving, don’t include the file source.